subiquity/controllers/filesystem.py

```diff
@@ -752,12 +752,9 @@ def guided_lvm(self, disk, lvm_options=None):
             spec['password'] = lvm_options['luks_options']['password']
         # create volume group on partition
         vg = self.create_volgroup(spec)
-        target_size = dehumanize_size("4G")
-        if target_size > vg.free_for_partitions:
-            target_size = int(vg.size*0.8)
         self.create_logical_volume(
             vg=vg, spec=dict(
-                size=target_size,
+                size=vg.size*8//10,
```

Comment:
Given that this is on LVM, I think preallocating 80% to the rootfs is excessive.  The user can always expand the rootfs later as necessary, but shrinking in order to use the space for something else can't be done online (best case: create a new volume that's <= 20%, copy rootfs data to the other volume, delete the original volume).

I think the default size should be the greater of (50% of VG, 2x the minimum required to copy the squashfs), with provisions for the case where the total VG space is < 2x the squashfs size.

2x gives reasonable "default" headroom for logs, package upgrades, new package installations, etc; and the user can and should adjust to taste from there based on their requirements.

But if you have a 2TB VG, allocating 80% of it to the rootfs is pretty ridiculous in terms of why you use LVM at all.

Though anyway, ZFS solves all of this so it would be more useful to more people in the long term to get the ZFS root story haken out.

Reply:
% only makes sense if there is a high/low watermarks.

If disk is less than 10GB, use all of it.
If disk is between 10GB and 20GB, use 10GB.
If disk is between 20GB and 200GB, use 50%.
If disk is larger than 200GB, use 100GB.

Thus we use upto 10GB, on disks upto 20GB in size.
Then we go to 50%, but cap out at 100GB.

Because 1TB / 2TB / 1PB disks do not need or want root to be one lvm volume..... Or at least so I hope.

Can we get info from like bootstrap how the deploy things? or maas?